daemon/endpoint.go

```diff
@@ -681,7 +684,31 @@ func (h *putEndpointIDLabels) Handle(params PutEndpointIDLabelsParams) middlewar
 }
 
 // OnIPIdentityCacheChange is called whenever there is a change of state in the
-// IPCache (pkg/ipcache). Currently does nothing as of now.
-func (d *Daemon) OnIPIdentityCacheChange() {
-
+// IPCache (pkg/ipcache).
+func (d *Daemon) OnIPIdentityCacheChange(modType ipcache.CacheModification, ipIDPair identity.IPIdentityPair) {
+	// TODO - see if we can factor this into an interface under something like
+	// pkg/datapath instead of in the daemon directly so that the code is more
+	// logically located.
+
+	// Update BPF Maps.
+	key := remotelxcmap.NewEndpointKey(ipIDPair.IP)
+	value := remotelxcmap.RemoteEndpointInfo{SecurityIdentity: uint16(ipIDPair.ID)}
+
+	switch modType {
+	case ipcache.Upsert:
+		err := remotelxcmap.RemoteLXCMap.Update(key, value)
```

Comment:
Please add a FIXME referring https://github.com/cilium/cilium/issues/3161 so we fix this up. We need to get rid of these one time attempts with logging as the only fallback